test/chainsaw-test.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: verify-extension
+spec:
+  timeouts:
+    apply: 5s
+    assert: 3m
+    delete: 30s
+  description: Verify an extension is properly installed
+  steps:
+  - name: Create a Cluster with the extension
+    try:
+    - apply:
+        file: cluster.yaml
```

Comment:
If we need to have an extension which has to be in shared_preload_libraries, should this conditionally use either `cluster.yaml` or `cluster-with-shared-preload-library`?

Reply:
> That would also be easier if it could be e.g. set in the image (as OCI annotation) and read by the operator.

Why would we need that if we are actually the ones producing this information directly at the source? Extending the image catalogue to cover these cases for the extensions will solve the problem in a static and declarative way, controlled at the source.

Moreover, what you are asking requires CNPG to access a registry, and I am not sure we want that (that's not our job, IMHO).

Reply:
Regarding the customization of each extension's GUCS, I've done a first implementation in https://github.com/cloudnative-pg/postgres-extensions-containers/pull/3/commits/764a19d88ca42b0fd8350ff377b6cc9eaed45f8f, and added a way to run more specific tests in case you define them in a `test/` dir in the extension's directory. Ideally I'd also like to automate generating the chainsaw `values.yaml` file locally, to facilitate running the tests locally, but IMHO that's an improvement to be done in a further issue..